engine/src/system.h

```diff
@@ -188,7 +188,7 @@ class MCMemoryFileHandle: public MCSystemFileHandle
 		// If there isn't enough room, extend
 		if (m_pointer + p_length > m_capacity)
```

Comment:
Not particularly. C does automatic upward promotion of integer types (as makes perfect sense) and warnings come about (generally) when you are casting a larger integer type to a smaller one (e.g. signed -> unsigned, int64 -> int32). The above line is just a comparison, no result is stored anywhere so auto promotion is absolutely fine.